vm/dependencies.go

```diff
@@ -27,33 +26,34 @@ import (
-	StateSyncServerDelay             time.Duration
-	ParsedBlockCacheSize             int
-	AcceptedBlockWindow              int
-	AcceptedBlockWindowCache         int
-	ContinuousProfilerConfig         profiler.Config
-	TargetBuildDuration              time.Duration
-	ProcessingBuildSkip              int
-	TargetGossipDuration             time.Duration
-	BlockCompactionFrequency         int
+	TraceConfig                      trace.Config    `json:"traceConfig"`
+	MempoolSize                      int             `json:"mempoolSize"`
+	AuthVerificationCores            int             `json:"authVerificationCores"`
+	VerifyAuth                       bool            `json:"verifyAuth"`
+	RootGenerationCores              int             `json:"rootGenerationCores"`
+	TransactionExecutionCores        int             `json:"transactionExecutionCores"`
+	StateFetchConcurrency            int             `json:"stateFetchConcurrency"`
+	MempoolSponsorSize               int             `json:"mempoolSponsorSize"`
+	StreamingBacklogSize             int             `json:"streamingBacklogSize"`
+	StateHistoryLength               int             `json:"stateHistoryLength"`               // how many roots back of data to keep to serve state queries
+	IntermediateNodeCacheSize        int             `json:"intermediateNodeCacheSize"`        // how many bytes to keep in intermediate cache
+	StateIntermediateWriteBufferSize int             `json:"stateIntermediateWriteBufferSize"` // how many bytes to keep unwritten in intermediate cache
+	StateIntermediateWriteBatchSize  int             `json:"stateIntermediateWriteBatchSize"`  // how many bytes to write from intermediate cache at once
+	ValueNodeCacheSize               int             `json:"valueNodeCacheSize"`               // how many bytes to keep in value cache
+	AcceptorSize                     int             `json:"acceptorSize"`                     // how far back we can fall in processing accepted blocks
+	StateSyncParallelism             int             `json:"stateSyncParallelism"`
+	StateSyncMinBlocks               uint64          `json:"stateSyncMinBlocks"`
+	StateSyncServerDelay             time.Duration   `json:"stateSyncServerDelay"`
+	ParsedBlockCacheSize             int             `json:"parsedBlockCacheSize"`
+	AcceptedBlockWindow              int             `json:"acceptedBlockWindow"`
+	AcceptedBlockWindowCache         int             `json:"acceptedBlockWindowCache"`
+	ContinuousProfilerConfig         profiler.Config `json:"continuousProfilerConfig"`
+	TargetBuildDuration              time.Duration   `json:"targetBuildDuration"`
+	ProcessingBuildSkip              int             `json:"processingBuildSkip"`
+	TargetGossipDuration             time.Duration   `json:"targetGossipDuration"`
+	BlockCompactionFrequency         int             `json:"blockCompactionFrequency"`
+	// Config is defined by the Controller
+	Config map[string]any `json:"config"`
```

Comment:
Decided to not make this `[]byte` because it makes the UX weird... the downside is that we have to re-marshal this to give `[]byte` to the implementing VM but I think this is fine since it's only done once upon initialization.

Reply:
Not sure how this will go later since what we're doing is non-standard, but I think this is alright for now. We can find a better approach if it causes issues.